core/src/main/java/org/polypheny/db/catalog/impl/AdapterRestore.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019-2023 The Polypheny Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.catalog.impl;
+
+import io.activej.serializer.annotations.Deserialize;
+import io.activej.serializer.annotations.Serialize;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import org.polypheny.db.adapter.Adapter;
+import org.polypheny.db.catalog.entity.allocation.AllocationCollection;
+import org.polypheny.db.catalog.entity.allocation.AllocationEntity;
+import org.polypheny.db.catalog.entity.allocation.AllocationGraph;
+import org.polypheny.db.catalog.entity.allocation.AllocationTable;
+import org.polypheny.db.catalog.entity.physical.PhysicalEntity;
+
+public class AdapterRestore {
```

Comment:
Why is this a class?

Reply:
Adapters are free to handle their AdapterCatalogs as they please, with some minor limitations that they have to expose their physical and allocation objects as well as their mapping. This class serves as a simple streamlined data holder to serialize this information in a less cluttered manner to preserve it for restart and hand it back to the adapters if required, to they can rebuild their internal `AdapterCatalog`.